engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/TypeDefinitionSet.scala

```diff
@@ -0,0 +1,54 @@
+package pl.touk.nussknacker.engine
+
+import cats.data.{NonEmptyList, Validated}
+import cats.data.Validated.{Invalid, Valid}
+import org.springframework.expression.spel.SpelNode
+import pl.touk.nussknacker.engine.api.expression.ExpressionParseError
+import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+import pl.touk.nussknacker.engine.definition.{DefinitionExtractor, ProcessDefinitionExtractor, TypeInfos}
+import pl.touk.nussknacker.engine.spel.TypedNode
+import pl.touk.nussknacker.engine.spel.ast.SpelAst.RichSpelNode
+
+
+class TypeDefinitionSet(typeDefinitions: Map[String, TypeInfos.ClazzDefinition]) {
+
+  def validateTypeReference(spelNode: SpelNode): Validated[NonEmptyList[ExpressionParseError], SpelNode] = {
+
+    val spelNodeChildAST = spelNode.children.headOption.getOrElse(throw new Exception("SpelNode has no children")).toStringAST
+
+    if(typeDefinitions.contains(spelNodeChildAST)) {
```

Comment:
I've missed one thing - spell need to have full class name or simple class name if you defined global imports for that class. I think that we need to replace `contains` with `find` invoked on `Set` and make checking global imports aware.

Reply:
I wonder, maybe it would be easier to just evaluate this node (using getValue) and obtain "real" class?

Reply:
Implemented it. Thanks